usdt-impl/src/common.rs

```diff
@@ -42,7 +42,11 @@ pub fn generate_type_check(types: &[dtrace_parser::DataType]) -> TokenStream {
 // registers.
 pub fn construct_probe_args(types: &[dtrace_parser::DataType]) -> (TokenStream, TokenStream) {
     // TODO this will fail with more than 6 parameters.
-    let abi_regs = ["rdi", "rsi", "rdx", "rcx", "r8", "r9"];
+    let abi_regs = if cfg!(any(target_arch = "arm64", target_arch = "aarch64")) {
+        ["r0", "r1", "r2", "r3", "r4", "r5"]
+    } else {
+        ["rdi", "rsi", "rdx", "rcx", "r8", "r9"]
```

Comment:
now that I think about it, it might warrant a comment here:
```rust
// x86-64 passes the first 6 function parameters in registers; subsequent parameters are passed on the stack. We currently limit ourselves to 6 probe parameters as navigating stack-passed parameters with the current semantics of the asm! macro would be challenging
```

Also what do you think about:

```rust
let abi_regs = if cfg!(target_arch = "x86_64") {
...
} else if cfg!(any(target_arch = "arm64", target_arch = "aarch64")) {
...
} else {
// Suddenly the Dungeon collapses!! - You die...
// i.e. a compiler error
}
```

Reply:
Good call, thank you!